tests/conditional_compilation/conftest.py

```diff
@@ -171,3 +190,85 @@ def save_session_info(pytestconfig, artifacts):
     """Fixture function for saving additional attributes to configuration file."""
     yield
     write_session_info(path=artifacts / SESSION_INFO_FILE, data=pytestconfig.session_info)
+
+
+@pytest.fixture(scope="session")
+def omz_path(request):
+    """Fixture function for command-line option."""
+    omz_path = request.config.getoption("omz_repo", skip=True)
+    validate_path_arg(omz_path, is_dir=True)
+
+    return omz_path
+
+
+@pytest.fixture(scope="session")
+def omz_cache_dir(request):
+    """Fixture function for command-line option."""
+    omz_cache_dir = request.config.getoption("omz_cache_dir", skip=True)
+    if omz_cache_dir:
+        try:
+            validate_path_arg(omz_cache_dir, is_dir=True)
+        except ValueError:
+            log.warning(f'The Open Model Zoo cache directory "{omz_cache_dir}" does not exist and will be created.')
+
+    return omz_cache_dir
+
+
+@pytest.fixture(scope="function")
+def prepared_models(openvino_ref, models, omz_path, omz_cache_dir):
+    """
+    Process models: prepare Open Model Zoo models, skip non-OMZ models.
+    """
+    for model in models:
+        if model.get("type") == "omz":
+            model["path"] = prepare_omz_model(openvino_ref, model, omz_path, omz_cache_dir)
+    models = [model["path"] for model in models]
+    return models
+
+
+def prepare_omz_model(openvino_ref, model, omz_path, omz_cache_dir):
```

Comment:
if the function computes new path for the model, path like `path: ${OMZ_MODELS_PATH}/densenet-201` from config isn't used at all, and OMZ_MODELS_PATH shouldn't be used too

I'd suggest to use path from config to dump model there, and OMZ_MODELS_PATH env var can be used to provide dynamism when defining models/IRs place